db/migrations/000001_db_setup.up.sql

```diff
@@ -0,0 +1,178 @@
+CREATE TYPE
+    "order_status" AS ENUM (
+        'pending',
+        'paid',
+        'shipped',
+        'delivered',
+        'cancelled'
+    );
+
+CREATE TYPE
+    "coupon_type" AS ENUM (
+        'percentage',
+        'fixed',
+        'shipping'
+    );
+
+CREATE TYPE "role_type" AS ENUM ( 'admin', 'customer' );
+
+CREATE TABLE
+    "cart" (
+        "id" SERIAL PRIMARY KEY,
+        "user_id" INT NOT NULL,
+        "shop_id" INT NOT NULL
+    );
+
+CREATE TABLE
+    "cart_product" (
+        "cart_id" INT NOT NULL,
+        "product_id" INT NOT NULL,
+        "quantity" INT NOT NULL
+    );
+
+CREATE TABLE
+    "order_history" (
+        "id" SERIAL PRIMARY KEY,
+        "user_id" INT NOT NULL,
+        "shop_id" INT NOT NULL,
+        "shipment" INT NOT NULL,
+        "total_price" INT NOT NULL,
+        "status" order_status NOT NULL,
+        "created_at" TIMESTAMP NOT NULL DEFAULT NOW()
+    );
+
+CREATE TABLE
+    "order_detail" (
+        "order_id" INT NOT NULL,
+        "product_id" INT NOT NULL,
+        "product_version" INT NOT NULL,
+        "quantity" INT NOT NULL
+    );
+
+CREATE TABLE
+    "cart_coupon" (
+        "cart_id" INT NOT NULL,
+        "coupon_id" INT NOT NULL
+    );
+
+CREATE TABLE
+    "product" (
+        "id" SERIAL PRIMARY KEY,
+        "version" SERIAL,
+        "shop_id" INT NOT NULL,
+        "name" VARCHAR(255) NOT NULL,
+        "description" TEXT NOT NULL,
+        "price" DECIMAL(10, 2) NOT NULL,
+        "image_id" UUID NOT NULL,
+        "due_date" TIMESTAMP NOT NULL,
+        "stock" INT NOT NULL,
+        "sales" INT NOT NULL,
+        "enabled" BOOLEAN NOT NULL DEFAULT TRUE,
```

Comment:
if the only disable condition is by `due_date`, we can remove this column to improve normalization

Reply:
There is another condition that the seller remove its product. Since we keep the record, we might need this.